mobile-app/app/screens/WalletNavigator/screens/CreateWallet/PinConfirmationV2.tsx

```diff
@@ -0,0 +1,156 @@
+import { translate } from '@translations'
+import { WalletParamList } from '@screens/WalletNavigator/WalletNavigator'
+import { useLogger } from '@shared-contexts/NativeLoggingProvider'
+import { useWhaleApiClient } from '@shared-contexts/WhaleContext'
+import { WalletAddressIndexPersistence } from '@api/wallet/address_index'
+import { EncryptedProviderData } from '@defichain/jellyfish-wallet-encrypted'
+import { MAX_ALLOWED_ADDRESSES } from '@shared-contexts/WalletContext'
+import { CreateWalletStepIndicatorV2 } from '@components/CreateWalletStepIndicatorV2'
+import { PinTextInputV2 } from '@components/PinTextInputV2'
+
+type Props = StackScreenProps<WalletParamList, 'PinConfirmation'>
+
+export function PinConfirmationV2 ({ route }: Props): JSX.Element {
+  const logger = useLogger()
+  const { network } = useNetworkContext()
+  const { setWallet } = useWalletPersistenceContext()
+  const client = useWhaleApiClient()
+  const {
+    pin,
+    words,
+    type
+  } = route.params
+  const [newPin, setNewPin] = useState('')
+
+  const [invalid, setInvalid] = useState<boolean>(false)
+  const [spinnerMessage, setSpinnerMessage] = useState<string>()
+
+  function verifyPin (input: string): void {
+    if (input.length !== pin.length) {
+      return
+    }
+    if (input !== pin) {
+      setNewPin('')
+      setInvalid(true)
+      return
+    } else {
+      setInvalid(false)
+    }
+
+    const copy = { words, network, pin }
+    setSpinnerMessage(translate('screens/PinConfirmation', 'It may take a few seconds to secure and encrypt your wallet'))
+    setTimeout(() => {
+      MnemonicEncrypted.toData(copy.words, copy.network, copy.pin)
+        .then(async encrypted => {
+          await MnemonicStorage.set(words, pin)
+          if (type === 'restore') {
+            await discoverWalletAddresses(encrypted)
+          }
+          await setWallet(encrypted)
+        })
+        .catch(logger.error)
+    }, 50) // allow UI render the spinner before async task
+  }
+
+  async function discoverWalletAddresses (data: WalletPersistenceDataI<EncryptedProviderData>): Promise<void> {
+    const provider = await MnemonicEncrypted.initProvider(data, network, {
+      /**
+       * wallet context only use for READ purpose (non signing)
+       * see {@link TransactionAuthorization} for signing implementation
+       */
+      async prompt () {
+        throw new Error('No UI attached for passphrase prompting')
+      }
+    })
+    const wallet = await initJellyfishWallet(provider, network, client)
+
+    // get discovered address
+    const activeAddress = await wallet.discover(MAX_ALLOWED_ADDRESSES)
+
+    // sub 1 from total discovered address to get address index of last active address
+    const lastDiscoveredAddressIndex = Math.max(0, activeAddress.length - 1)
+    await WalletAddressIndexPersistence.setLength(lastDiscoveredAddressIndex)
+  }
+
+  return (
+    <ThemedScrollViewV2
+      dark={tailwind('bg-gray-900')}
```

Comment:
dont use v1 color, the default one from `ThemedScrollViewV2` should be correct, same thing for light mode